src/sbvr-api/sbvr-utils.ts

```diff
@@ -1166,9 +1166,13 @@ const getIdField = (
 		| 'resourceName'
 		| 'vocabulary'
 	>,
-) =>
-	// TODO: Should resolveSynonym also be using the finalAbstractSqlModel?
-	getFinalAbstractSqlModel(request).tables[resolveSynonym(request)].idField;
+) => {
+	const resolvedSynonym = resolveSynonym(request);
+	const translatedTable = getAbstractSqlModel(request).tables[resolvedSynonym];
+	const resourceName =
+		translatedTable != null ? translatedTable.resourceName : resolvedSynonym;
```

Comment:
Why do we expect `translatedTable` to be null at any point?

Reply:
My understanding is that it will be `nullish` when it is in the DB model (it doesn't translate to anything)